src/main.rs

```diff
@@ -205,37 +206,56 @@ fn git_diff(path: &Path) -> Result<String, clap::error::Error> {
 
 // Takes the `&str` output of `git_diff` above, and filters out irrelevant
 // lines. Cannot be a part of `git_diff` because this returns a vector of string
-// slices (for efficiency) on top of strings allocated inside of `git_diff`.
-fn sanitized_diff_lines(diff: &str) -> Vec<&str> {
+// slices (for efficiency) and their line location on top of strings allocated
+// inside of `git_diff`.
+fn sanitized_diff_lines(diff: &str) -> Vec<(usize, &str)> {
+    let mut line_number = 0;
+    let hunk_header_regex = Regex::new(r"@@ -(\d+),(\d+) \+(\d+),(\d+) @@").unwrap();
+
     diff.split('\n')
         .enumerate()
-        // Strip the first 5 version control lines, and only consider lines
-        // prefixed with "+" that are more than one character long.
-        .filter(|&(i, line)| i > 4 && line.starts_with('+') && line.len() > 1)
-        // Remove the "+" version control prefix.
-        .map(|(_, line)| &line[1..])
+        .filter_map(|(i, line)| {
+            // Extract line number from hunk header
+            if let Some(captures) = hunk_header_regex.captures(line) {
+            // Start line number count at position 3 of hunk header
+            // which will indicate where in file we start counting lines
+                line_number = captures[3].parse::<usize>().unwrap(); 
+            }
+            // Excludes version control lines
+            if i > 5 {
```

Comment:
Instead of nesting everything below inside this condition, I think it would be more readable to invert it. Basically if `i <= 5` just early-return `None`. Then everything below can be de-dented by 4 spaces.